src/averageAges.js

```diff
@@ -14,12 +24,15 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menList = people.filter((person) => person.sex === MALE);
+  const menFromCentury = menList
+    .filter((man) => Math.ceil(man.died / CENTURY) === century);
+
+  return (!century)
+    ? getAverengeAge(menList)
+    : getAverengeAge(menFromCentury);
```

Comment:
If `century` will have a false value, then you will be doing extra filtering. 
Move the check of this value to the `menFromCentury` filtering, so as not to make an unnecessary loop through the array.